packages/pages/src/dev/server/ssr/generateTestData.ts

```diff
@@ -52,16 +55,30 @@ export const generateTestDataForSlug = async (
     vite,
     templateFilepaths
   );
-  const featuresConfig = getTemplatesConfig(templateModuleCollection);
-  const featuresConfigForEntityPages: FeaturesConfig = {
-    features: featuresConfig.features.filter((f) => "entityPageSet" in f),
-    streams: featuresConfig.streams,
-  };
-  const args = getCommonArgs(featuresConfigForEntityPages, projectStructure);
-  args.push("--slug", slug);
 
-  const parsedData = await spawnTestDataCommand(stdout, "yext", args);
-  return getDocumentByLocale(parsedData, locale);
+  const allParsedData: any[] = [];
+  templateModuleCollection.forEach(async (templateModuleInternal) => {
+    const featureConfig = convertTemplateConfigToFeatureConfig(
+      templateModuleInternal.config
+    );
+    const streamConfig = convertTemplateConfigToStreamConfig(
+      templateModuleInternal.config
+    );
+    const featuresConfigForEntityPages: FeaturesConfig = {
+      features: "entityPageSet" in featureConfig ? [featureConfig] : [],
+      streams: streamConfig ? [streamConfig] : [],
+    };
+    const args = getCommonArgs(featuresConfigForEntityPages, projectStructure);
+    args.push("--slug", templateModuleInternal.config.slugField ?? slug);
```

Comment:
`slug` represents the slug value, which is coming from the url request. It's not "what slug field to use." We'll need new functionality in the CLI function.